From a2cce81137333d53b57a5fe2fd46e6e35b92c5ec Mon Sep 17 00:00:00 2001
From: Greg Kaiser <gkaiser@google.com>
Date: Wed, 1 Aug 2018 12:47:48 -0700
Subject: [PATCH 3/4] Revert "Support Speck encryption."

This is a partial revert of
commit 6a0fa62dc71157f85b24ed15e39ada277f9b5201.

Remove the Speck encryption support. It was eventually
decided not to allow Speck in Android P, so this code
is no longer needed and wasn't used outside of testing.

We retain the e4crypt_get_policy_flags() changes from original
commit, though.

Bug: 112009351
Test: Confirmed AES continues to work with FBE.
Change-Id: I53b091dd4c5392de7a00a8c168919fb55e527eaf
(cherry picked from commit fa9ce0e4a240d50fde168369c2095d1e14241a97)
---
 ext4_utils/ext4_crypt.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/ext4_utils/ext4_crypt.cpp b/ext4_utils/ext4_crypt.cpp
index 5ec8723b..36fe11f7 100644
--- a/ext4_utils/ext4_crypt.cpp
+++ b/ext4_utils/ext4_crypt.cpp
@@ -52,8 +52,6 @@ struct ext4_encryption_policy {
 
 #define EXT4_ENCRYPTION_MODE_AES_256_XTS    1
 #define EXT4_ENCRYPTION_MODE_AES_256_CTS    4
-#define EXT4_ENCRYPTION_MODE_SPECK128_256_XTS 7
-#define EXT4_ENCRYPTION_MODE_SPECK128_256_CTS 8
 #define EXT4_ENCRYPTION_MODE_AES_256_HEH    126
 #define EXT4_ENCRYPTION_MODE_PRIVATE        127
 
@@ -260,8 +258,6 @@ int e4crypt_policy_ensure(const char *directory, const char *policy,
     if (!strcmp(contents_encryption_mode, "software") ||
         !strcmp(contents_encryption_mode, "aes-256-xts")) {
         contents_mode = EXT4_ENCRYPTION_MODE_AES_256_XTS;
-    } else if (!strcmp(contents_encryption_mode, "speck128/256-xts")) {
-        contents_mode = EXT4_ENCRYPTION_MODE_SPECK128_256_XTS;
     } else if (!strcmp(contents_encryption_mode, "ice")) {
         contents_mode = EXT4_ENCRYPTION_MODE_PRIVATE;
     } else {
@@ -272,8 +268,6 @@ int e4crypt_policy_ensure(const char *directory, const char *policy,
 
     if (!strcmp(filenames_encryption_mode, "aes-256-cts")) {
         filenames_mode = EXT4_ENCRYPTION_MODE_AES_256_CTS;
-    } else if (!strcmp(filenames_encryption_mode, "speck128/256-cts")) {
-        filenames_mode = EXT4_ENCRYPTION_MODE_SPECK128_256_CTS;
     } else if (!strcmp(filenames_encryption_mode, "aes-256-heh")) {
         filenames_mode = EXT4_ENCRYPTION_MODE_AES_256_HEH;
     } else {
-- 
2.17.1

